### PR TITLE
New tutorial: Building and publishing an OCaml library

### PR DESCRIPTION
**Pull request by johnwhitington**
This is a draft of a PR which adds a tutorial on building a library with dune, testing it, and publishing it to opam.

The equivalent PR in the opam-repository is https://github.com/ocaml/opam-repository/pull/18513

To quote that PR:

"It would be great if the content & style could be scrutinized by the experts, so I do not mislead beginners (I don't yet use dune myself, in particular)."

There are also probably things which could be added, such as formal tests rather than just an example.

### CODE REVIEW

**Comment by johnwhitington** [comment]
(By the way, if anyone knows the magic git command to stop my unrelated change from another PR showing up in all the new PRs I create from new branches, please enlighten me..)

**Comment by johnwhitington** [comment]
NB: must add documentation to `roman.mli` and have ocamldoc / odoc process it, perhaps so its visible online too?

**Comment by johnwhitington** [comment]
I have updated the tutorial (and its example on OPAM) to show how one can use `opam-publish` more than once to publish subsequent updates.

This PR is now ready for review.

**Comment by braibant** [comment]
> (By the way, if anyone knows the magic git command to stop my unrelated change from another PR showing up in all the new PRs I create from new branches, please enlighten me..)

Well, I don't know if it's helpful, but it looks like https://github.com/ocaml/ocaml.org/pull/1532/commits/cfaf93aef5551b48d44653c8d12ba7121bcecbb5 is part of this pull request (sitting in branch `reorder-short` in your fork), but has not been merged into `ocaml:master` yet. It also looks like you are picking up merge commits along the way (are you using `git pull` without `--rebase` by any chance?).

My guess here is that the history of this PR might be more legible if you were rebasing it by doing something along the lines of https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request , but I am not sure what is the merge / rebase etiquette that we use in this repo.

**Comment by johnwhitington** [comment]
Thanks @braibant. I followed the instructions on that page, and it seemed to do the trick.

This PR is now ready for review. Once approved, I can add a commit which adds it to the table of contents.

File: site/learn/tutorials/building_and_publishing.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+`dune-project` file to read:
+
+```scheme
+(lang dune 2.8)
+(name roman)
+(generate_opam_files true)
+(source (github johnwhitington/roman))
+(license BSD-2-Clause)
+(authors "John Whitington")
+(maintainers "contact@coherentgraphics.co.uk")
```
Comment (avsm) [inline site/learn/tutorials/building_and_publishing.md]: I would recommend not putting your own author details in the example. A remarkable number of people just cut-and-paste, and so using "example" ones (c.f. RFC2606) just makes it explicit that that default data provided isn't canonical.

**Review by avsm — APPROVED** [review]
Lgtm for the final changes, aside from a minor comment.

**Comment by johnwhitington** [comment]
Done. This can be merged now. I'm going to add it to the index page when fixing https://github.com/ocaml/ocaml.org/pull/1565 or in a separate PR. Including index page updates in these individual PRs is confusing.

**Comment by johnwhitington** [comment]
#1565 now has resolved conflicts. So once it is merged, I can add the index page fix to this PR. So please merge 1565 first, and not this PR until I have added the index entry.

File: site/learn/tutorials/building_and_publishing.md (hunk 1 above)
Comment (johnwhitington) [reply site/learn/tutorials/building_and_publishing.md]: Done.

**Comment by johnwhitington** [comment]
Closed in favour of https://github.com/ocaml/ocaml.org/pull/1606

**Pull request closed without merging**